src/zabapgit_syntax_highlighter.prog.abap

```diff
@@ -0,0 +1,762 @@
+          lv_css_class TYPE string.
+
+    FIELD-SYMBOLS:
+          <match> TYPE ty_match.
+
+    LOOP AT it_matches ASSIGNING <match>.
+      lv_chunk = substring( val = iv_line off = <match>-offset len = <match>-length ).
+
+      CASE <match>-token.
+        WHEN c_token-keyword.
+          lv_css_class = c_css-keyword.
+        WHEN c_token-comment.
+          lv_css_class = c_css-comment.
+        WHEN c_token-text.
+          lv_css_class = c_css-text.
+        WHEN c_token-none.
+          CLEAR: lv_css_class.
+      ENDCASE.
+
+      lv_chunk = me->apply_style( iv_line  = lv_chunk
+                                  iv_class = lv_css_class ).
+
+      rv_line = rv_line && lv_chunk.
+    ENDLOOP.
+
+  ENDMETHOD.                    "format_line
+
+  METHOD apply_style.
+
+    DATA lv_escaped TYPE string.
+
+    lv_escaped = escape( val = iv_line  format = cl_abap_format=>e_html_attr ).
+    IF iv_class IS NOT INITIAL.
+      rv_line = |<span class="{ iv_class }">{ lv_escaped }</span>|.
+    ELSE.
+      rv_line = lv_escaped.
+    ENDIF.
+
+  ENDMETHOD.                    "apply_style
+
+  METHOD process_line.
+
+    DATA: lt_matches TYPE ty_match_tt.
+
+    IF strlen( iv_line ) = 0.
+      RETURN.
+    ENDIF.
+
+    lt_matches = me->parse_line( iv_line ).
+
+    me->order_matches( EXPORTING iv_line    = iv_line
+                       CHANGING  ct_matches = lt_matches ).
+
+    rv_line = me->format_line( iv_line    = iv_line
+                               it_matches = lt_matches ).
+
+  ENDMETHOD.                    " process_line
+
+ENDCLASS.                       " lcl_code_highlighter IMPLEMENTATION
+
+
+*----------------------------------------------------------------------*
+*       CLASS ltcl_code_highlighter definition
+*----------------------------------------------------------------------*
+CLASS ltcl_code_highlighter DEFINITION FINAL
+  FOR TESTING RISK LEVEL HARMLESS DURATION SHORT
+  INHERITING FROM cl_aunit_assert.
+
+  PRIVATE SECTION.
+
+    DATA:
+          mo  TYPE REF TO lcl_code_highlighter.
+
+    METHODS: setup.
+    METHODS: process_line     FOR TESTING.
+    METHODS: parse_and_order  FOR TESTING.
+    METHODS: format_line      FOR TESTING.
+    METHODS: apply_style      FOR TESTING.
+
+ENDCLASS.                       " ltcl_code_highlighter
+*----------------------------------------------------------------------*
+*       CLASS ltcl_code_highlighter IMPLEMENTATION
+*----------------------------------------------------------------------*
+CLASS ltcl_code_highlighter IMPLEMENTATION.
+
+  DEFINE _generate_matches.
+    ls_match-token    = &1.
+    ls_match-offset   = &2.
+    ls_match-length   = &3.
+    ls_match-text_tag = &4.
+    IF &5 = 'PARSE'.
+      APPEND ls_match to lt_after_parse.
+    ELSEIF &5 = 'ORDER'.
+      APPEND ls_match to lt_after_order.
+    ENDIF.
+  END-OF-DEFINITION.
+
+  METHOD setup.
+    CREATE OBJECT mo.
+  ENDMETHOD.                    " setup
+
+  METHOD parse_and_order.
```

Comment:
Got it. Maybe it's a matter of preference. I prefer to isolate each test case in a separate test method. more on the why [here](http://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test)